README.md

```diff
@@ -1 +1,85 @@
-# node-arguments
\ No newline at end of file
+# Node Arguments
+
+## Overview
+
+This lesson will cover the Node command-line arguments and how to access them.
+
+## Objectives
+
+1. Describe how the CLI argument input to Node script works
+1. Describe an argument array
+2. Describe first arguments
+
+## CLI Arguments
+
+How do we pass data to Node? `node` is just a command-line program and we can pass arguments to it. For example, we can pass two numbers to a program (file `sum.js`) that calculates the sum of them:
+
+```
+$ node sum.js 10 47
+```
+
+The result will be 57. We can pass numbers or strings. This approach is useful when you're running scripts which require small enough input that you can fit in in the command line, or whin building command-line tools with Node. Another use caes is when you have a complex program that needs to be launched in different modes, e.g., development mode with more verbose logging.
+
+## Accessing Arguments
+
+You've learned how to pass arguments, but how can you access them when you write your own program instead of using existing programs like sum.js? It's straightforward. All you need is to access `process.argv`. It's an array so all the Array methods available. 
+
+Consider the sum.js program. I'm using `shift` to get rid of the first two elements and then `forEach()` to iterate through indefinite number of arguments. I convert each argument to a number and add them up to the `sum` variable:
+
+```js
+var sum = 0
+process.argv.shift()
+process.argv.shift()
+process.argv.forEach(function summarize(item, index, list) {
+  sum += parseInt(item, 10)
+})
+console.log(sum)
+```
+
+## First Arguments
+
+The first arguments of the array were `node` and `sum` or the command and the script name. This is exactly what we had in the command line when we were running the program:
+
+```
+$ node sum 10 47
+```
+
+Note: The `.js` extension is optional. Node will try to match the filename to what's in the folder. If there's a `sum.js` file, it will run it.
+
+The exact `process.argv` array will look like (on my machine):
+
+```
+[ '/usr/local/bin/node',
+  '/Users/azat/Documents/Code/learn-co/node-arguments/sum',
+  '10',
+  '47' ]
+```
+
+
+Just for fun, take a look at this command in which we use `-e` to execute code right away without saving it to a file:
+
+```
+$ node -e "console.log(process.argv)" 10 47
```

Comment:
most of the top npm and JavaScript modules use`$` in their docs. this is considered the best practice for docs because it tell readers the difference between code and CLI command

![screenshot 2016-02-12 19 16 50](https://cloud.githubusercontent.com/assets/1137314/13025468/30d1777e-d1bd-11e5-9101-2d904d269e92.png)


https://www.npmjs.com/package/pm2
https://www.npmjs.com/package/forever
https://www.npmjs.com/package/express
https://www.npmjs.com/package/browserify
https://www.npmjs.com/package/bower